lxd/storage/drivers/driver_zfs_utils.go

```diff
@@ -31,8 +31,17 @@ const (
 	zfsMaxVolBlocksize = 128 * 1024
 )
 
+func (d *zfs) isBlockBacking(vol Volume) bool {
```

Comment:
OK so I think we need to have an `isBlockBacked(vol) bool` function in `driver_common` that checks for:

```go
driver.Info().BlockBacking 
```

And then update `volume.IsBlockBacked()` to call `v.driver.Info().isBlockBacked(v)` rather than `driver.Info().BlockBacking `.

That way we can then have this ZFS specific function just check the "zfs.block_mode" setting.

We may as well rename this to `isBlockBacked` to to align with the `vol.IsBlockBacked()` function.

We should keep the check for `|| v.mountFilesystemProbe` in `vol.IsBlockBacked()` though, so it will end up looking like this:

```go
// IsBlockBacked indicates whether storage device is block backed.
func (v Volume) IsBlockBacked() bool {
	return v.driver.Info().isBlockBacked(v) || v.mountFilesystemProbe
}
```

Otherwise we are likely going to hit issues with `volume.IsBlockBacked()` using the current ZFS pool "zfs.block_mode" setting rather than only looking at the volume.